### ISSUE TITLE
s3fs is not working with Minio

### ISSUE THREAD

**Issue by rusher81572** [minio/minio#4070]
Hello, I tried s3fs on Raspbian, Ubuntu, and MacOS and I can not see any objects:

To Reproduce:

```
root@rpi-1:~# apt-get install s3fs --force-yes -y
Reading package lists... Done
Building dependency tree
Reading state information... Done
The following NEW packages will be installed:
  s3fs
0 upgraded, 1 newly installed, 0 to remove and 3 not upgraded.
Need to get 0 B/202 kB of archives.
After this operation, 425 kB of additional disk space will be used.
WARNING: The following packages cannot be authenticated!
  s3fs
Selecting previously unselected package s3fs.
(Reading database ... 50663 files and directories currently installed.)
Preparing to unpack .../s3fs_1.80-2~bpo8+1_armhf.deb ...
Unpacking s3fs (1.80-2~bpo8+1) ...
Processing triggers for man-db (2.7.0.2-5) ...
Setting up s3fs (1.80-2~bpo8+1) ...
```
(Create a bucket called html on Minio then proceed)

```
s3fs html /test -o passwd_file=/mnt/passwd,url=http://minio-server:8000
```

```
mount

s3fs on /test type fuse.s3fs (rw,nosuid,nodev,relatime,user_id=0,group_id=0)
```

```
ls /test

(nothing)
```

```
cp /etc/passwd /test 

cp: cannot stat ‘/test/passwd’: Input/output error
```

**Comment by vadmeste** [comment]
@rusher81572, can you test with Minio master? it has a change which enhances support of s3fs.

**Comment by rusher81572** [comment]
@vadmeste It is worse than before now because it won't mount:

Syslog:

```
Apr 07 12:50:33 rpi-1 s3fs[16631]: s3fs.cpp:s3fs_check_service(3779): unable to connect - result of checking service.
```

```
root@rpi-1:~/work/bin#  s3fs html ./foo -o passwd_file=passwd,url=http://192.168.0.101:8010;ls ./foo
ls: cannot access ./foo: Software caused connection abort
```

**Comment by vadmeste** [comment]
@rushenn you need to add this parameter `-o use_path_request_style`, please test and update us.

**Comment by rusher81572** [comment]
That worked, thanks

**Comment by rusher81572** [comment]
And it worked for the current minio releae

**Comment by deekoder** [comment]
Can we close this as working as intended?

**Comment by harshavardhana** [comment]
Fixed with #3897, will be available in the next release.  

We can perhaps add a sample document for using `s3fs` as well indicating usage of `-o use_path_request_style` . // cc @NitishT @koolhead17

**Comment by koolhead17** [comment]
sure @harshavardhana
